### ISSUE TITLE
DOI/HDL format in suggested citations

### ISSUE THREAD

**Issue by adam3smith** [IQSS/dataverse#4237]
This is following up on the exchange with @mercecrosas here: https://github.com/IQSS/dataverse/issues/2438#issuecomment-334532204 but since it's not specific to file citations, creating a separate ticket.

Currently, the suggested citation format produced by Dataverse prints the DOI as doi:10.1234/DVN/56789
This is in line with what used to be common display norms for DOIs (e.g., at the time of the seminal Altman&King 2007 data citation article), but it is no longer recommended. All relevant agencies, other major data repositories, and citation manuals now use/require display of DOIs in URL form, i.e https://doi.org/10.1234/DVN/56789 
*Registration Agencies*
Crossref: https://www.crossref.org/display-guidelines/
DataCite: https://support.datacite.org/v1.1/docs/datacite-doi-display-guidelines

*Data Repositories*
Dryad: http://datadryad.org/resource/doi:10.5061/dryad.3278c
ICPSR: http://doi.org/10.3886/E100603V1
QDR: https://doi.org/10.5064/F6Z31WJ1  (and many more like Pangea, UK Data, Gesis, etc.)

*Citation Manuals*
Chicago Manual: http://www.chicagomanualofstyle.org/tools_citationguide/citation-guide-2.html#cg-journal
APA Manual: http://blog.apastyle.org/apastyle/digital-object-identifier-doi/

It'd be terrific if you could adjust that -- one less thing for us to customize. Incidentally, looking at old, Dataverse 3 screenshots, you seem to have used the URL format previously.

**Comment by djbrooke** [comment]
Thanks @adam3smith.

@scolapasta - is this a duplicate of or similar to the issue we talked about other day (#898)? If so, let's consolidate under this issue.

**Comment by adam3smith** [comment]
This is different -- #898/#3583 is about the structure of the DOI itself, this is about how the DOI is _displayed_

**Comment by djbrooke** [comment]
Got it, thanks!

**Comment by pameyer** [comment]
@adam3smith - one thing that I'm confused about with these recommendations is that they appear to make it more difficult for systems hand treat a DOI as an identifier, as opposed to a link (aka - the vast majority of links do not translate back to a persistent identifier).  Do you happen to know of any resources I should look at to reduce my confusion?

**Comment by adam3smith** [comment]
I don't know of any good resources, but the way I see this, these are _display_ guidelines designed for human readers/clickers. The DOI as an identifier should be (and is, for Dataverse as ` <meta name="DC.identifier" content="doi:10.7910/DVN/HVONZL" />`) in machine-readable metadata so that software handling identifiers can act on it.

**Comment by pameyer** [comment]
That makes sense - but the mention of bots/crawlers made me thing systems rather than people.  Thanks!

**Comment by adam3smith** [comment]
right -- but that (I assume you're referring to the CrossRef guidlines) refers to general purpose bots like regular google search crawlers -- we want them to treat DOIs as links for pageRank and similar purposes.

Where bots specialize and act on PIDs, they should first look for and be served more sophisticated metadata with the identifier clearly marked as such -- the doi:10.1234/5678 notation in plaintext still requires processing the page via regex, which is possible but never ideal.

**Comment by mercecrosas** [comment]
I agree with Sebastian, it should be in the form of:

https://doi.org/10.1234/DVN/56789
<https://urldefense.proofpoint.com/v2/url?u=https-3A__doi.org_10.1234_DVN_56789&d=DwMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=n9HCCtgqDPssu5vpqjbO3q4h2g6vMeTOp0Ez7NsdVFM&m=9jHxW6Dlnjw7PcxM5r3yK-tRTunwASwpzB3LAFsdBsw&s=v1PevnSsukEpqy72fq0haruFQ0A9pm_IxMpt6HDXPPg&e=>

I specified that to Tania last week, in the latest revisions, but I think I
was not clear enough about its importance.

Merce

On Wed, Oct 25, 2017 at 7:58 PM, Sebastian Karcher <notifications@github.com

**Comment by LauraHuisintveld** [comment]
Does this also count for the display of handles?
One participating university at DataverseNL asked if we could display the handle as a link. 
(It was noticed that when users copy and past the whole citation, the URL is lost)
So, instead of 'hdl:10411/OF7UCA', the citation should show 'http://hdl.handle.net/10411/OF7UCA' .

(I couldn't find any display guidelines specifically for handles)

**Comment by sbarbosadataverse** [comment]
If you hover over the handle/DOI, you will notice that the link is there,
and includes the http://
<https://urldefense.proofpoint.com/v2/url?u=http-3A__hdl.handle.net_10411_OF7UCA&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=OzsMKtdJPY4QIgCSlHr4U-1YKV9tieMqe3Qz86WxVY8&e=>
It's just not displayed in the citation.

On Mon, Feb 19, 2018 at 9:41 AM, LauraHuisintveld <notifications@github.com>
wrote:

> Does this also count for the display of handles?
> One participating university at DataverseNL asked if we could display the
> handle as a link.
> (It was noticed that when users copy and past the whole citation, the URL
> is lost)
> So, instead of 'hdl:10411/OF7UCA', the citation should show '
> http://hdl.handle.net/10411/OF7UCA
> <https://urldefense.proofpoint.com/v2/url?u=http-3A__hdl.handle.net_10411_OF7UCA&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=OzsMKtdJPY4QIgCSlHr4U-1YKV9tieMqe3Qz86WxVY8&e=>'
> .
>
> (I couldn't find any display guidelines specifically for handles)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IQSS_dataverse_issues_4237-23issuecomment-2D366712473&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=BF_mGbJfoMZyhihboR9wGaW5e02w3Z_OmrUfoC589ys&e=>,
> or mute the thread
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AH79Kvc34CDB7Z-5FmFBkGunOZiCEH2N1bks5tWYgugaJpZM4QGEYY&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=o6nQe73q5G-ybpJQpo-vncij6wRS5bvDCBYRJjEW2Ug&e=>
> .
>



-- 
Sonia Barbosa
Manager of Data Curation, The Dataverse Project
Manager of the Murray Research Archive, IQSS
Data Science
Harvard University

All Harvard Dataverse Repository inquiries should be sent to:
support@dataverse.harvard.edu
All software inquiries should be sent to: support@dataverse.org


Dataverse 4.0 is now available for use!
http://dataverse.harvard.edu

All test dataverses should be created  in 4.0 Demo!
https://demo.dataverse.org/

Join our Dataverse Community!
https://groups.google.com/forum/#!forum/dataverse-community

**Comment by sbarbosadataverse** [comment]
When I copy and paste the entire citation into word, everything is
preserved.

On Mon, Feb 19, 2018 at 12:40 PM, Barbosa, Sonia <sbarbosa@g.harvard.edu>
wrote:

> If you hover over the handle/DOI, you will notice that the link is there,
> and includes the http://
> <https://urldefense.proofpoint.com/v2/url?u=http-3A__hdl.handle.net_10411_OF7UCA&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=OzsMKtdJPY4QIgCSlHr4U-1YKV9tieMqe3Qz86WxVY8&e=>
> It's just not displayed in the citation.
>
> On Mon, Feb 19, 2018 at 9:41 AM, LauraHuisintveld <
> notifications@github.com> wrote:
>
>> Does this also count for the display of handles?
>> One participating university at DataverseNL asked if we could display the
>> handle as a link.
>> (It was noticed that when users copy and past the whole citation, the URL
>> is lost)
>> So, instead of 'hdl:10411/OF7UCA', the citation should show '
>> http://hdl.handle.net/10411/OF7UCA
>> <https://urldefense.proofpoint.com/v2/url?u=http-3A__hdl.handle.net_10411_OF7UCA&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=OzsMKtdJPY4QIgCSlHr4U-1YKV9tieMqe3Qz86WxVY8&e=>'
>> .
>>
>> (I couldn't find any display guidelines specifically for handles)
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_IQSS_dataverse_issues_4237-23issuecomment-2D366712473&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=BF_mGbJfoMZyhihboR9wGaW5e02w3Z_OmrUfoC589ys&e=>,
>> or mute the thread
>> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AH79Kvc34CDB7Z-5FmFBkGunOZiCEH2N1bks5tWYgugaJpZM4QGEYY&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=8R6PzVqt1PEocauQgZMGXsGz29-nb19M7eqlo1d8EVs&m=7L-HEPKbvBjwZQc3S6xd96WWS8mYcJ4MUzqS7fRrEBA&s=o6nQe73q5G-ybpJQpo-vncij6wRS5bvDCBYRJjEW2Ug&e=>
>> .
>>
>
>
>
> --
> Sonia Barbosa
> Manager of Data Curation, The Dataverse Project
> Manager of the Murray Research Archive, IQSS
> Data Science
> Harvard University
>
> All Harvard Dataverse Repository inquiries should be sent to:
> support@dataverse.harvard.edu
> All software inquiries should be sent to: support@dataverse.org
>
>
> Dataverse 4.0 is now available for use!
> http://dataverse.harvard.edu
>
> All test dataverses should be created  in 4.0 Demo!
> https://demo.dataverse.org/
>
> Join our Dataverse Community!
> https://groups.google.com/forum/#!forum/dataverse-community
>



-- 
Sonia Barbosa
Manager of Data Curation, The Dataverse Project
Manager of the Murray Research Archive, IQSS
Data Science
Harvard University

All Harvard Dataverse Repository inquiries should be sent to:
support@dataverse.harvard.edu
All software inquiries should be sent to: support@dataverse.org


Dataverse 4.0 is now available for use!
http://dataverse.harvard.edu

All test dataverses should be created  in 4.0 Demo!
https://demo.dataverse.org/

Join our Dataverse Community!
https://groups.google.com/forum/#!forum/dataverse-community

**Comment by LauraHuisintveld** [comment]
Yes, you are right, that works fine. (I Should have tried that self first...)
When asked for a persistent identifier, some users copy only the handle part out of the citation and send it to the publisher. (This also includes the URL). But it is possible that someone will type it over manually, and thus loose the URL.
Not a big problem indeed. Won't happen that often I suppose.

I was just wondering if the display guidelines for DOI's (displaying DOI as a link) would also be applicable for handles. (I think so)

I think the issue here is more about this:

This is from the Crossref FAQ about display guidelines (https://www.crossref.org/display-guidelines/#FAQ):

" Why does the DOI have to be displayed as a link on the page that the DOI links to?

Some members have reported resistance from colleagues to displaying the Crossref DOI on the landing page as a link (they say the DOI link in that location appears superfluous as it appears to link to itself). The Crossref DOI must be displayed as a link, because it is both an identifier and a persistent link. It is easier for users when members display the DOI as a full link as they can copy it easily. In addition, many users don’t know what a DOI is but they know what a link is. We want to encourage the DOI to be used as a persistent link, and to be shared and used in other applications (e.g. reference managers). A fully linked DOI wherever it appears, enables this."

**Comment by jggautier** [comment]
In the suggested citations of datasets in Dataverse 3.x, handles are displayed with http as well (http://hdl.handle.net/11272/10579). And this is how DSpace handles the display of handles (https://wiki.duraspace.org/display/DSPACE/PersistentIdentifiers).

@adam3smith, I'd like to rename this issue to account for handles as well, if that's okay.

**Comment by adam3smith** [comment]
of course, please go ahead.

**Comment by pdurbin** [comment]
I just wanted to mention that in pull request #4470 we are switching DOIs from http to https and removing "dx.doi".

**Comment by matthew-a-dunlap** [comment]
I've switched over our display representation of hdl/doi for the dataset page and search results (https://github.com/IQSS/dataverse/tree/4237-doi-hdl-display-format).

Extra notes:
- I was unable to completely fake a handle dataset on my development machine. I was able to confirm my fix worked for doi & handles in the search results, but make sure to confirm both on the dataset page (both Related Publication in top metadata / the url in the blue "header" box).
- I don't see the purpose at this point in asking for the "ID Number" for a related publication. Maybe I misunderstand the field but don't we already have that data from the url?
- I'm new to this area of the code so I may have missed a place where we present the handle/doi urls
